packages/palette/src/elements/Collapse/Collapse.tsx

```diff
@@ -1,35 +1,102 @@
 import React from "react"
-import { animated, Spring } from "react-spring"
 
 export interface CollapseProps {
-  /** Determines whether content is expanded or collapsed */
   open: boolean
 }
-
-/** Collapses content with animation when open is not true */
+/**
+ * Collapse component for the web
+ */
 export class Collapse extends React.Component<CollapseProps> {
-  state = {
-    mounted: false,
+  wrapperModifyTimeout: ReturnType<typeof setTimeout>
+  wrapperRef: HTMLDivElement | null = null
+
+  onTransitionEnd = (ev: TransitionEvent) => {
+    if (!this.wrapperRef) {
+      return
+    }
+    // when animating open, we set the wrapper's height to an explicit pixel
+    // value. When the transition ends we want to set the height back to 'auto'
+    // so that if the content of the wrapper changes it's height will change too
+    if (ev.propertyName === "height") {
+      this.wrapperRef.style.height = this.props.open ? "auto" : "0px"
+    }
   }
 
   componentDidMount() {
-    this.setState({ mounted: true })
+    if (!this.wrapperRef) {
+      return
+    }
+
+    this.wrapperRef.addEventListener("transitionend", this.onTransitionEnd)
+  }
+
+  componentDidUpdate() {
+    if (!this.wrapperRef) {
+      return
+    }
+    if (this.props.open && this.wrapperRef.style.height !== "auto") {
+      // animate opening
+      // measure goal height
+      const prevHeight = this.wrapperRef.style.height || "0px"
+      this.wrapperRef.style.height = "auto"
+      const goalheight = this.wrapperRef.offsetHeight
+      this.wrapperRef.style.height = prevHeight
+      // wait for a tick before setting goal height to allow transition
+      this.wrapperModifyTimeout = setTimeout(() => {
+        this.wrapperRef.style.height = goalheight + "px"
+      }, 10)
```

Comment:
Why `setTimeout(…, 10)` over `setImmediate(…)` if the need is to perform in on the next tick of the runloop?

Reply:
`setImmediate` is not on the standards track any more. https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate. `requestAnimationFrame` might be the most conceptually 'correct' to use here, but `setTimeout` works and was chosen through muscle memory more than anything else.

Reply:
Good to know!